enhancements/api-review/add-new-CRD-ImageDigestMirrorSet-and-ImageTagMirrorSet-to-config.openshift.io.md

```diff
@@ -0,0 +1,429 @@
+`imagecontentsourcepolicy.operator.openshift.io/v1alpha1` will stay 12 months or 3 releases from the announcement of deprecation, whichever is longer. During the upgrade path,
+if an 4.11 upgrade fails mid-way through, or if the 4.11 cluster is
+misbehaving, the user can rollback to the version that supports `ImageContentSourcePolicy`.
+Their `ImageDigestMirrorSet` and `ImageTagMirrorSet` dependent workflows will be clobbered and broken if rollback to a version
+that lacks support for these CRDs. They can still configure the `ImageContentSourcePolicy` to use mirrors and keep previous behavior.
+
+### Version Skew Strategy
+
+Upgrade skew will not impact this feature. The MCO does not require skew check. CRI-O with n-2 OpenShift skew will still be able to handle the new property.
+
+### Operational Aspects of API Extensions
+
+Impacts of API extensions:
+- Components that must be upgraded to use ImageDigestMirrorSet:
+  - [Installer](https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/pkg/asset/manifests/imagesourcepolicy.go#L18)
+  - [Hypershift](https://github.com/openshift/hypershift/blob/baceec23098d39af089b06c503425c3bbee554d3/control-plane-operator/controllers/hostedcontrolplane/manifests/imagecontentsource.go)
+  - [MCO](https://github.com/openshift/machine-config-operator) The machine-config-operator/pkg/controller/container-runtime-config controller needs to operate the ImageDigestMirrorSet and ImageTagMirrorSet CRDs.
+  - [Image-registry](https://github.com/openshift/image-registry/blob/a87e6c50cd973723de8b5471453de7c345403d56/pkg/dockerregistry/server/simplelookupicsp.go#L60)
+  - [Cluster-config-operator](https://github.com/openshift/cluster-config-operator)
+  - [Openshift-api-server](https://github.com/openshift/openshift-apiserver/blob/98786f917ffc7d3dc3b05893f405970b87a419b9/pkg/image/apiserver/registries/registries.go)
+  - [Runtime utils](https://github.com/openshift/runtime-utils/blob/8b8348d80d1d1e7b6cf06fb009d5965e0b55baa2/pkg/registries/registries.go#L50)
+  - [Openshift-controller-manager](https://github.com/openshift/openshift-controller-manager/blob/2a11f145ad7fcf3e92460800de1d13ba7fbb90b0/pkg/build/controller/build/build_controller.go#L20943)
+
+The Node team will be making changes to the above impacted components to enable the new CRDs. Planning target CY21Q4
+Jira card  [OCPNODE-717](https://issues.redhat.com/browse/OCPNODE-717) will record the upgrade of ImageDigestMirrorSet.
+
+#### Failure Modes
+
+Failure of applying new ImageDigestMirrorSet and ImageTagMirrorSet should not impact the cluster health.
+If there are existing ImageDigestMirrorSet, ImageTagMirrorSet or ImageContenSourcePolicy objects, they
+should continue to work.
+OCP Node team is likely to be called upon in case of escalation with one of the failure modes.
+
+#### Support Procedures
+
+If the cluster fails to apply new CRs, the condition of the object will be `Failed`.
```

Comment:
Of which object? The MirrorSet CRs don’t have a status field at all, do they? So is this talking about the MCO operator / the MCC controller?

Reply:
I forgot this CR does not have status. I should just mention the MCC logs in this part.